src/components/common/EmojiPicker.js

```diff
@@ -1,130 +1,97 @@
-  '😯',
-  '😲',
-  '😳',
-  '🥺',
-  '😦',
-  '😧',
-  '😨',
-  '😰',
-  '😥',
-  '😢',
-  '😭',
-  '😱',
-  '😖',
-  '😣',
-  '😞',
-  '😓',
-  '😩',
-  '😫',
-  '🥱',
-  '😤',
-];
-//Use this to set the maximum number of emojis a user can submit
-const emojiLimit = 6;
-const Emoji = props => {
-  const { emoji, handleClick } = props;
-  return (
-    <button className="Emoji" onClick={() => handleClick(emoji)}>
-      {emoji}
-    </button>
-  );
-};
+import { emojiList, emojiLimit } from '../../utils/constants';
+import emojiIcon from '../../assets/icons/emoji.svg';
+
 const EmojiPicker = props => {
   const { getChildState } = props;
   const [selectedEmojis, setSelectedEmojis] = useState([]);
+  const [keyboardVisible, setKeyboardVisible] = useState(false);
+
   //If parent component wants to get this state, pass cb into props. Converts Array to String of emojis to match backend data type.
   useEffect(() => {
     const selectedEmojisString = selectedEmojis.join('') + ',';
     if (getChildState) {
       getChildState(selectedEmojisString);
     }
   }, [selectedEmojis, getChildState]);
-  const handleAddEmoji = emoji => {
+
+  const handleToggleEmoji = emoji => {
     if (selectedEmojis.length < emojiLimit && !selectedEmojis.includes(emoji)) {
       setSelectedEmojis([...selectedEmojis, emoji]);
+    } else if (selectedEmojis.includes(emoji)) {
+      handleRemoveEmoji(emoji);
     }
   };
+
   const handleRemoveEmoji = emoji => {
     setSelectedEmojis(
       selectedEmojis.filter(selectedEmoji => selectedEmoji !== emoji)
     );
   };
+
+  const handleKeyboardVisible = () => {
+    setKeyboardVisible(!keyboardVisible);
+  };
+
   return (
-    <div className="EmojiPicker">
-      <Card title="Give Feedback" className="emoji-feedback-card">
-        <div>
-          {selectedEmojis.map(emoji => (
-            <Emoji emoji={emoji} handleClick={handleRemoveEmoji} />
-          ))}
+    <div className="emoji-picker">
+      {keyboardVisible && (
+        <div className="emoji-keyboard-position">
+          <div className="emoji-keyboard-container">
+            <div className="emoji-keyboard">
+              {emojiList.map(emoji => (
+                <Emoji
+                  emoji={emoji}
+                  selectedEmojis={selectedEmojis}
+                  handleClick={handleToggleEmoji}
+                />
+              ))}
+            </div>
+            <div className="done-button-container">
+              <button
+                className="done-button font-size-32"
+                onClick={handleKeyboardVisible}
+              >
+                Done
+              </button>
+            </div>
+          </div>
         </div>
-      </Card>
-      <Card className="emoji-selection-card">
-        {emojiList.map(emoji => (
-          <Emoji emoji={emoji} handleClick={handleAddEmoji} />
+      )}
+      <div className="selected-emojis">
+        {selectedEmojis.map(emoji => (
+          <Emoji emoji={emoji} handleClick={handleRemoveEmoji} />
```

Comment:
Did react not throw a warning for not having a `key` with this?  I'd assume we need one

Reply:
Good catch, just got buried under all the other warnings still present. This is why I prefer TS :+1:

Reply:
Same, I've been missing TS lol